code/model/modelread.cpp

```diff
@@ -3362,14 +3359,7 @@ int subobj_find_2d_bound(float radius ,matrix * /*orient*/, vec3d * pos,int *x1,
 	if (pnt.flags & PF_OVERFLOW)
 		return 2;
 
-	t = (width * Canv_w2)/pnt.world.xyz.z;
-	w = t*Matrix_scale.xyz.x;
-
-	t = (height*Canv_h2)/pnt.world.xyz.z;
-	h = t*Matrix_scale.xyz.y;
-
-	// Use the smaller of the width and height (forces the resulting targetting reticle to be square on non-square aspect ratios).
-	w = h = (w < h) ? w : h;
+	w = h = radius * Canv_h2 * Matrix_scale.xyz.y / pnt.world.xyz.z;
```

Comment:
I recommend breaking this up into a few more lines to show that we're operating on the height component first, and then copying over to the width.  Also recommend making the Matrix_scale a seperate operation from the projection calculation to make it clearer as to whats going on.

Reply:
The matrix scale is part of the projection. However, I broke it up a bit to aid in clarity. Any good code optimizing compiler should be able to reduce it to the same number of instructions anyway.